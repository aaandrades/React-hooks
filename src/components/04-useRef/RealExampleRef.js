import React, { useState } from "react";
import { MultipleCustomHooks } from "../03-examples/MultipleCustomHooks";
import "./useRef.css";

export const RealExampleRef = () => {
  const [show, setShow] = useState(false);
  return (
    <div>
      <h3>Real Example Ref</h3>
      {show && <MultipleCustomHooks />}
      <br></br>
      <br></br>
      <button className="btn btn-secondary" 
                onClick={() => setShow(!show)}>
        Show/Hide
      </button>
    </div>
  );
};
