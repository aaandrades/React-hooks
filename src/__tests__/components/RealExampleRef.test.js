import React from "react";
import { shallow } from "enzyme";
import { RealExampleRef } from "../../components/04-useRef/RealExampleRef";

describe("Tests of RealExampleRef", () => {
  const wrapper = shallow(<RealExampleRef />);

  it("should show correctly", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(false);
  });

  it("should show the component MultipleCustomHooks", () => {
    wrapper.find("button").simulate("click");
    expect(wrapper.find("MultipleCustomHooks").exists()).toBe(true);
  });
});
