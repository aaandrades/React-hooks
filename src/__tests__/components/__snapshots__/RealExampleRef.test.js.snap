// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of RealExampleRef should show correctly 1`] = `
<div>
  <h3>
    Real Example Ref
  </h3>
  <br />
  <br />
  <button
    className="btn btn-secondary"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
